### ISSUE TITLE
torch.einsum() is not supported?

### ISSUE THREAD

**Issue by henbucuoshanghai** [Tencent/TNN#1362]
my pytorch  model has torch.einsum()   ,i converted  it to onnx and tnn successfully,
but i load the tnn model has error ?????
like bellows?

E/tnn: Forward [File /home/ps/TNN-master/source/tnn/core/default_network.cc][Line 600] Forward error code: 0x4000 msg: einsum() fewer operands were provided than specified in the equation, exit
E/tnn: Forward [File /home/ps/TNN-master/source/tnn/core/default_network.cc][Line 600] Forward error code: 0x4000 msg: einsum() fewer operands were provided than specified in the equation, exit

**Comment by 1627180283** [comment]
模型转换的时候，转换命令里加上 -align 可以成功转换吗

**Comment by 1627180283** [comment]
模型里使用到的 einsum 可以用 matmul 或者其他算子等效替代吗？使用 einsum 可能性能会差一些

**Comment by henbucuoshanghai** [comment]
torch.einsum()就一句命令&nbsp; &nbsp;

**Comment by henbucuoshanghai** [comment]
&nbsp; &nbsp; def forward(self, x, A):
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp; x = self.conv(x)

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp; n, kc, t, v = x.size()
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp; x = x.view(n, self.kernel_size, kc // self.kernel_size, t, v)
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp; x = torch.einsum('nkctv,kvw-&gt;nctw', (x, A))

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp; return x.contiguous(), A

**Comment by 1627180283** [comment]
torch.einsum('nkctv,kvw->nctw', (a, b)) 应该可以用这个替代。

def replace_einsum(a, b):
    out = torch.sum(torch.matmul(a, b.view(1, b.size(0), 1, b.size(1), b.size(2))), dim=1)

因为我这边不知道两个输入的 shape，我用 shape 是 (1, 50, 30, 45, 25) 和 (50, 25, 10) 的数据测了一下，这个输入在 pytorch 里跑相较于直接用 einsum 有 1.34 的加速比。但是拆开之后也会有其他开销，有的情况不一定都会有加速收益，具体要看你输入数据的 shape。

**Comment by henbucuoshanghai** [comment]
**
[my.zip](https://github.com/Tencent/TNN/files/7336493/my.zip)
**

**Comment by henbucuoshanghai** [comment]
onnx输入是 1,3,32,26,1 维度&nbsp; 输出是1,10 维度




用onnxruntime推理&nbsp; &nbsp;得到结果&nbsp; 但是速度为何如此慢呢？安装onnxruntime-gpu&nbsp; 没用。
请问怎么回事呢

**Comment by 1627180283** [comment]
![image](https://user-images.githubusercontent.com/37806708/137108504-06e9d286-7b75-422b-bba7-31199a3a2c17.png)
我zhe'b这边看到的消息好像没有正常显示

**Comment by henbucuoshanghai** [comment]
最后打扰一次 请问not align对tnn模型影响大吗？
